tritium-proxy/src/main/java/com/palantir/tritium/proxy/Proxies.java

```diff
@@ -58,41 +56,26 @@ private Proxies() {
      *
      * @param iface primary interface class
      * @param delegateClass delegate class
-     * @param additionalInterfaces additional interfaces
      * @return the set of interfaces for the specified classes
      * @throws IllegalArgumentException if the specified interface class is not an interface
      */
     static Class<?>[] interfaces(Class<?> iface,
-                                 Class<?> delegateClass,
-                                 Collection<Class<?>> additionalInterfaces) {
+                                 Class<?> delegateClass) {
         checkIsInterface(iface);
-        Objects.requireNonNull(additionalInterfaces, "additionalInterfaces");
         Objects.requireNonNull(delegateClass, "delegateClass");
 
         Set<Class<?>> interfaces = new LinkedHashSet<>();
         interfaces.add(iface);
-        interfaces.addAll(additionalInterfaces);
         if (delegateClass.isInterface()) {
             interfaces.add(delegateClass);
+        } else {
+            interfaces.addAll(Arrays.asList(delegateClass.getInterfaces()));
         }
```

Comment:
I don't think this change makes a difference aside from making me feel better, all current callers use `delegate.getClass()` and I'm not aware of any way for Object.getClass to return an interface.